packages/js/test-env/src/index.ts

```diff
@@ -163,8 +163,8 @@ export async function buildAndDeployApi({
   );
 
   // create a new ENS domain
-  const domainName = ensName ?? generateName();
-  const apiEns = `${domainName}.eth`;
+  const apiEns = ensName ?? `${generateName()}.eth`;
+  const domainName = apiEns.split(".").slice(0, -1).join(".");
```

Comment:
I don't know if this would work with a domain like `coingecko.defiwrapper.eth`... can we make a test with this type of ENS domain to see if it works?

Reply:
In order to unblock this, we can merge this PR now, and implement subdomain support in a new PR @namesty